tools/cldr-apps/js/src/esm/cldrLocales.mjs

```diff
@@ -1,11 +1,56 @@
 /*
  * cldrLocales: encapsulate functions concerning locales for Survey Tool
  */
+import * as cldrAjax from "./cldrAjax.mjs";
 import * as cldrEvent from "./cldrEvent.mjs";
 import * as cldrLoad from "./cldrLoad.mjs";
+import { LocaleMap } from "./cldrLocaleMap.mjs";
 import * as cldrMenu from "./cldrMenu.mjs";
+import * as cldrNotify from "./cldrNotify.mjs";
+import * as cldrRetry from "./cldrRetry.mjs";
 import * as cldrStatus from "./cldrStatus.mjs";
 import * as cldrSurvey from "./cldrSurvey.mjs";
+import * as cldrText from "./cldrText.mjs";
+
+// The current locale may be "USER" temporarily, meaning the back end should choose an appropriate locale
+// for the current user. The real locale ID should be set when a server response contains it.
+const USER_LOCALE_ID = "USER";
+
+const VALIDATE_ALL_LOCALES = true;
+const VALIDATE_ALL_LOCALES_VERBOSE = false;
+
+const LOCALE_REGEX = new RegExp("^[a-zA-Z0-9_]*$");
+
+let didValidateAll = false;
+
+async function fetchMap() {
+  const url = getUrl();
+  return await cldrAjax
+    .doFetch(url)
+    .then(cldrAjax.handleFetchErrors)
+    .then((r) => r.json())
+    .then(setMap)
+    .catch(console.error);
+}
+
+function getUrl() {
+  const p = new URLSearchParams();
+  p.append("what", "menus"); // cf. WHAT_GET_MENUS in SurveyAjax.java
```

Comment:
WHAT_GET_MENUS is now used completely separately for getting locales and for getting menus. It could be separated (and modernized to api) on the back end. Or, possibly it should be recombined for fewer http requests, however that would require refactoring of the menu code since that's very crufty with `cldrLoad.myLoad` and `flipper`